src/hooks/useDropzoneWithFileReader.ts

```diff
@@ -0,0 +1,50 @@
+import { DropzoneOptions, DropzoneProps, useDropzone } from 'react-dropzone';
+
+import {
+  DEFAULT_MIME_TYPES,
+  DEFAULT_MAX_FILE_SIZE,
+} from '~shared/FileUpload/limits';
+import { HandleFileAccept } from '~shared/FileUpload/types';
+import getFileReader from '~utils/fileReader';
+
+const useDropzoneWithFileReader = ({
```

Comment:
This hook makes the use of dropzone more flexible and separates the logic from a specific view.